solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,679 @@
+    error_message = "'kms_encryption_enabled_bucket' can not be true if 'enable_vpc_flow_logs' is set to false."
+  }
+
+  validation {
+    condition     = var.enable_vpc_flow_logs && var.kms_encryption_enabled_bucket ? ((var.existing_flow_logs_bucket_kms_key_crn != null || var.existing_kms_instance_crn != null) ? true : false) : true
+    error_message = "Either 'existing_flow_logs_bucket_kms_key_crn' or 'existing_kms_instance_crn' is required if 'enable_vpc_flow_logs' and 'kms_encryption_enabled_bucket' are set to true."
+  }
+}
+
+variable "skip_cos_kms_iam_auth_policy" {
+  type        = bool
+  description = "To skip creating an IAM authorization policy that allows Cloud Object Storage(COS) to access KMS key."
+  default     = false
+}
+
+variable "management_endpoint_type_for_bucket" {
+  description = "The type of endpoint for the IBM Terraform provider to use to manage Cloud Object Storage buckets (`public`, `private`, or `direct`). If you are using a private endpoint, make sure that you enable virtual routing and forwarding (VRF) in your account, and that the Terraform runtime can access the IBM Cloud Private network."
+  type        = string
+  default     = "direct"
+  validation {
+    condition     = contains(["public", "private", "direct"], var.management_endpoint_type_for_bucket)
+    error_message = "The specified `management_endpoint_type_for_bucket` is not valid. Specify a valid type of endpoint for the IBM Terraform provider to use to manage Cloud Object Storage buckets."
+  }
+}
+
+variable "cos_bucket_class" {
+  type        = string
+  default     = "standard"
+  description = "The storage class of the newly provisioned Cloud Object Storage bucket. Specify one of the following values for the storage class: `standard`, `vault`, `cold`, `smart` (default), or `onerate_active`."
+  validation {
+    condition     = contains(["standard", "vault", "cold", "smart", "onerate_active"], var.cos_bucket_class)
+    error_message = "Specify one of the following values for the `cos_bucket_class`: `standard`, `vault`, `cold`, `smart`, or `onerate_active`."
+  }
+}
+
+variable "force_delete" {
+  type        = bool
+  description = "Whether to delete all the objects in the flow logs Cloud Object Storage bucket before the bucket is deleted."
+  default     = true
+}
+
+variable "add_bucket_name_suffix" {
+  type        = bool
+  description = "Add a randomly generated suffix that is 4 characters in length, to the name of the newly provisioned Cloud Object Storage bucket. Do not use this suffix if you are passing the existing Cloud Object Storage bucket. To manage the name of the Cloud Object Storage bucket manually, use the `flow_logs_cos_bucket_name` variables."
+  default     = true
+}
+
+variable "flow_logs_cos_bucket_archive_days" {
+  description = "The number of days before the `archive_type` rule action takes effect for the flow logs cloud object storage bucket."
+  type        = number
+  default     = 20
```

Comment:
curious where this number came from?

Reply:
I took this number from the default value defined in module - https://github.com/terraform-ibm-modules/terraform-ibm-cos/blob/main/modules/buckets/variables.tf#L36